arch/eventloop_posix_udp.c

```diff
@@ -36,6 +44,524 @@ typedef struct {
+     * a temporary integer here. */
+    UA_Int32 enable = enableLoopback;
+#if UA_IPV6
+    if(UA_setsockopt(sockfd,
+                     ai_family == AF_INET6 ? IPPROTO_IPV6 : IPPROTO_IP,
+                     ai_family == AF_INET6 ? IPV6_MULTICAST_LOOP : IP_MULTICAST_LOOP,
+                     (const char *)&enable,
+                     sizeof (enable)) < 0)
+#else
+        if(UA_setsockopt(sockfd, IPPROTO_IP, IP_MULTICAST_LOOP,
+                     (const char *)&enable,
+                     sizeof (enable)) < 0)
+#endif
+    {
+        UA_LOG_SOCKET_ERRNO_WRAP(
+            UA_LOG_ERROR(logger, UA_LOGCATEGORY_NETWORK,
+                         "PubSub Connection creation failed. Loopback setup failed: "
+                         "Cannot set socket option IP_MULTICAST_LOOP. Error: %s",
+                         errno_str));
+        return UA_STATUSCODE_BADINTERNALERROR;
+    }
+    return UA_STATUSCODE_GOOD;
+}
+
+static UA_INLINE void
+setTimeToLive(UA_SOCKET sockfd,
+              UA_UInt32 messageTTL,
+              int ai_family,
+              const UA_Logger *logger) {
+    /* Set Time to live (TTL). Value of 1 prevent forward beyond the local network. */
+#if UA_IPV6
+    if(UA_setsockopt(sockfd,
+                     ai_family == PF_INET6 ? IPPROTO_IPV6 : IPPROTO_IP,
+                     ai_family == PF_INET6 ? IPV6_MULTICAST_HOPS : IP_MULTICAST_TTL,
+                     (const char *)&messageTTL,
+                     sizeof(messageTTL)) < 0)
+#else
+        if(UA_setsockopt(sockfd, IPPROTO_IP, IP_MULTICAST_TTL,
+                     (const char *)&messageTTL,
+                     sizeof(messageTTL)) < 0)
+#endif
+    {
+        UA_LOG_SOCKET_ERRNO_WRAP(
+            UA_LOG_WARNING(logger, UA_LOGCATEGORY_NETWORK,
+                           "PubSub Connection creation problem. Time to live setup failed: "
+                           "Cannot set socket option IP_MULTICAST_TTL. Error: %s",
+                           errno_str));
+    }
+}
+
+static UA_INLINE void
+setReuseAddress(UA_SOCKET sockfd, UA_Boolean enableReuse, const UA_Logger *logger) {
+    /* Set reuse address -> enables sharing of the same listening address on
+    * different sockets */
+    if(enableReuse){
+        int enableReuseVal = 1;
+        if(UA_setsockopt(sockfd, SOL_SOCKET, SO_REUSEADDR,
+                         (const char*)&enableReuseVal, sizeof(enableReuseVal)) < 0) {
+            UA_LOG_SOCKET_ERRNO_WRAP(
+                UA_LOG_WARNING(logger, UA_LOGCATEGORY_NETWORK,
+                               "PubSub Connection creation problem. Reuse address setup failed: "
+                               "Cannot set socket option SO_REUSEADDR. Error: %s",
+                               errno_str));
+        }
+    }
+}
+
+#ifdef __linux__
+static UA_StatusCode
+setSocketPriority(UA_SOCKET sockfd, UA_UInt32 *socketPriority, const UA_Logger *logger) {/* Setting the socket priority to the socket */
+    if(socketPriority != NULL) {
```

Comment:
Early return instead of deep nesting.

Reply:
> Early return instead of deep nesting.

If condition can be left out anyway, because the caller asserts != NULL already